File: SWEA/D3/knapsack01.py
file = open("./SWEA/D3/Knapsack01.txt", "r")

input = file.readline

t = int(input())

for tc in range(1, t + 1) : 
	n, k = map(int, input().split())
	items = [list(map(int, input().split())) for _ in range(n)]

	dp = [[0 for _ in range(k + 1)] for _ in range(n + 1)]

	for i in range(1, n + 1) : 
		for j in range(1, k + 1) : 
			if items[i-1][0] <= j : 
				dp[i][j] = max(dp[i-1][j], items[i-1][1] + dp[i-1][j-items[i-1][0]])
			else : 
				dp[i][j] = dp[i-1][j]

	# for i in range(n + 1) : 
	# 	for j in range(k + 1) : 
	# 		print(dp[i][j], end = " ")
	# 	print()

	print("#%d %d" %(tc, dp[n][k]))

	
file.close()

File: SWEA/D3/Knapsack01.java
import java.util.Arrays;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Knapsack01 {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new FileReader("./SWEA/D3/Knapsack01.txt"));

		int t = Integer.parseInt(br.readLine());

		for (int tc = 1; tc <= t; tc++) {
			String[] line = br.readLine().split(" ");
			// System.out.println(Arrays.toString(line));
			int n = Integer.parseInt(line[0]); 
			int k = Integer.parseInt(line[1]);
			// System.out.println(n + " " + k);

		 	List<float[]> things = new ArrayList[n];

			for (int h = 0; h < n; h ++) {
				line = br.readLine().split(" ");
				int v = Integer.parseInt(line[0]);
				int c = Integer.parseInt(line[1]);
				things.append({c/v, v, c});
			}
		}

		br.close();
	}
}